File: Longest_Substring_Without_Repeat/Longest_Substring_Without_Repeat.py
def longest_substring_without_repeat(s):

    output_needed = []
    max_length = 0
    start = 0

    for i in range(len(s)):
        while s[i] in output_needed:
            output_needed.remove(s[start])
            start += 1

        output_needed.append(s[i])
        if len(output_needed) > max_length:
            max_length = len(output_needed)


    print(max_length)

longest_substring_without_repeat(s = "pwwkew")
longest_substring_without_repeat(s = "abcabcbb")
longest_substring_without_repeat(s = "bbbbb")

File: Longest_Substring_Without_Repeat/Longest_Substring_Without_Repeat.java

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import java.util.ArrayList;
import java.util.List;

public class Longest_Substring_Without_Repeat{
    public static void main(String[] args){
        String s = "abcabcbb";
        List<Character> output_needed = new ArrayList<>();
        int max_length = 0;
        int start = 0;

        for (int i = 0; i < s.length(); i++) {
            while (output_needed.contains(s.charAt(i))) {
                output_needed.remove((Character) s.charAt(start));
                start++;
            }

            output_needed.add(s.charAt(i));

            if (output_needed.size() > max_length) {
                max_length = output_needed.size();
            }
        }
        System.out.println(max_length);

    }
}
